prefix-map.c

```diff
@@ -35,40 +35,40 @@ static void init_prefix_map(struct prefix_map *prefix_map,
 static void add_prefix_item(struct prefix_map *prefix_map,
 			    struct prefix_item *item)
 {
-	struct prefix_map_entry *e = xmalloc(sizeof(*e)), *e2;
+	struct prefix_map_entry e = { { NULL } }, *e2;
 	int j;
 
-	e->item = item;
-	e->name = e->item->name;
+	e.item = item;
+	e.name = item->name;
 
-	for (j = prefix_map->min_length; j <= prefix_map->max_length; j++) {
-		if (!isascii(e->name[j])) {
-			free(e);
+	for (j = prefix_map->min_length;
+	     j <= prefix_map->max_length && e.name[j]; j++) {
+		/* Avoid breaking UTF-8 multi-byte sequences */
+		if (!isascii(e.name[j]))
```

Comment:
Does this mean that we never put names with utf8 chars into the map?  The way I read this is that if there is a utf8 char anywhere between min and max, we skip the name, right?

Reply:
We totally do.